ui/webui/src/components/app.jsx

```diff
@@ -40,10 +43,12 @@ export const Application = () => {
     const [isStorageReady, setIsStorageReady] = useState(false);
     const [notifications, setNotifications] = useState({});
     const [conf, setConf] = useState();
+    const [beta, setBeta] = useState();
     const { path } = usePageLocation();
 
     useEffect(() => cockpit.file("/run/anaconda/bus.address").watch(setAddress), []);
     useEffect(() => readConf().then(setConf, ex => console.error("Failed to parse anaconda configuration")), []);
+    useEffect(() => readBuildstamp().then(buildstamp => setBeta(!getIsFinal(buildstamp)), ex => console.error("Failed to parse anaconda configuration")), []);
```

Comment:
You can have one useEffect call for these.

``` js

useEffect(() => {
    cockpit.file("/run/anaconda/bus.address").watch(setAddress);
    readConf().then(setConf, ex => console.error("Failed to parse anaconda configuration");
    readBuildstamp().then(buildstamp => setBeta(!getIsFinal(buildstamp)), ex => console.error("Failed to parse anaconda configuration"));
}, []);
```

Please adjust, did not notice that there were already 2. Practically there is not difference, just less boilerplate code.

Reply:
Also try to avoid maybe ultra long lines.
I like to put the .then arguments in new lines; so for this.

``` js

useEffect(() => readBuildstamp().then(
    buildstamp => setBeta(!getIsFinal(buildstamp)),
    ex => console.error("Failed to parse anaconda configuration: ", ex.message))
, []);
```